skyportal/handlers/api/comment.py

```diff
@@ -31,8 +31,11 @@ def get(self, comment_id):
         comment = Comment.query.get(comment_id)
         if comment is None:
             return self.error('Invalid comment ID.')
-        # Ensure user/token has access to parent source
-        s = Source.get_if_owned_by(comment.source.id, self.current_user)
+        # Ensure user/token has access to parent source/candidate
```

Comment:
The changes to this file are due to the fact that the parent may now either be a comment or a candidate, so we add checks here